vespalib/src/vespa/vespalib/stllike/hashtable.h

```diff
@@ -186,39 +185,37 @@ class hashtable : public hashtable_base
         typedef const Value* pointer;
         typedef std::forward_iterator_tag iterator_category;
 
-        const_iterator(const hashtable * hash, next_t start) : _hash(start), _subNode(start), _hashTable(hash) {
-            advanceToNextValidHash();
-        }
-        const_iterator(const hashtable * hash, next_t start, next_t subNode) : _hash(start), _subNode(subNode), _hashTable(hash) { }
-        const_iterator(const iterator &i)
-            : _hash(i._hash), _subNode(i._subNode), _hashTable(i._hashTable) {}
-        const Value & operator * ()  const { return _hashTable->get(_subNode); }
-        const Value * operator -> () const { return & _hashTable->get(_subNode); }
-        const_iterator & operator ++ () {
-            if (_hashTable->_nodes[_subNode].hasNext()) {
-                _subNode = _hashTable->_nodes[_subNode].getNext();
-            } else {
-                _hash++;
+        const_iterator(const hashtable * hash) : _current(0), _hashTable(hash) {
+            if ((_current < _hashTable->initializedSize()) && ! _hashTable->_nodes[_current].valid()) {
```

Comment:
same as above, totally empty hash should give end iterator

Reply:
I did find tests that tested that implicit. I added explicit tests for the same in #4575.
I think it is fine as is.